internal/operator/terraform/files.go

```diff
@@ -159,106 +166,150 @@ resource "gardener_shoot" "test_cluster" {
 `
 )
 
-// Terraform is an Operator.
-type Terraform struct {
+// initClusterFiles initializes all necessary files for a cluster in the given data directory
+func initClusterFiles(dataDir string, p types.ProviderType, cfg map[string]interface{}) error {
+	dir, err := clusterDir(dataDir, cfg["project"].(string), cfg["cluster_name"].(string), p)
+	if err != nil {
+		return err
+	}
+
+	// create module file
+	var data []byte
+	switch p {
+	case types.GCP:
+		data = []byte(gcpClusterTemplate)
+	case types.Gardener:
+		t, err := expandGardenerClusterTemplate(cfg)
+		if err != nil {
+			return err
+		}
+		data = []byte(t)
+	case types.Azure:
+		data = []byte(azureClusterTemplate)
+	case types.AWS:
+		data = []byte(awsClusterTemplate)
+	}
+	if err := ioutil.WriteFile(filepath.Join(dir, tfModuleFile), data, 0700); err != nil {
+		return err
+	}
+
+	// create vars file
+	var vars strings.Builder
+	for k, v := range cfg {
+		switch t := v.(type) {
+		case int:
+			vars.WriteString(fmt.Sprintf("%s = \"%d\"\n", k, t))
```

Comment:
Might be good to check the returned error

Reply:
done thanks!